PocketKit/Sources/SharedPocketKit/Migrations/LegacyUserMigration.swift

```diff
@@ -68,18 +67,30 @@ public class LegacyUserMigration {
             throw LegacyUserMigrationError.missingData
         }
 
-        let legacyStore: LegacyStore
+        return decryptedData
+    }
+
+    private func getLegacyStore(from decryptedData: Data) throws -> LegacyStore {
         do {
-            legacyStore = try JSONDecoder().decode(LegacyStore.self, from: decryptedData)
+            return try JSONDecoder().decode(LegacyStore.self, from: decryptedData)
         } catch {
             throw LegacyUserMigrationError.failedDeserialization(error)
         }
+    }
+
+    @discardableResult
+    public func perform(migrationAnalytics: () -> Void) throws -> Bool {
```

Comment:
can we rename `migrationAnalytics`? If I understand correctly this is meant to launch a braze message instead of tracking analytics with SnowPlow?

Reply:
I wanted to keep "braze" out of LegacyUserMigration as much as possible.
`fireMigrationEvent`?

Reply:
I agree that we don't need to put the braze name: looks like that closure executes a pre-migration task, so whichever name resembles that task is fine for me.